src/super_gradients/common/deprecate.py

```diff
@@ -0,0 +1,61 @@
+import warnings
+from typing import Optional
+from functools import wraps
+
+
+def make_function_deprecated(version: str, new_function_name: Optional[str] = None, new_module_name: Optional[str] = None):
```

Comment:
For a long time, we have been using "Deprecated" library (which is not that good)
If we are adding our own depreciation mechanism, let's do it correctly:
1. Call the decorator @deprecate - that is the convention 
2. move this change to its ow PR
3. Params should be
  - deprecated_in_v: str   (i.e. 3.2.0)
  - removed_in_v: str        (i.e. 3.5.0)
  - target: runable               (i.e. a function) can be None
  - reason: str
4. add assertion: assert removed_in_v < __version__   (so we will remember to remove this function when releasing the right version)
5. note that the new function is not a name - it is the function itself. this is you don't need the module name. you can infer it.